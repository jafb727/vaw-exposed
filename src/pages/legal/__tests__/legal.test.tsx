/**
 * @author Jose A Felix
 * @name legal.test
 * @description Legal page component unit testing file
 */

/** -------------------------------------------------------- */

/** @import Components */
import Legal from "..";

/** @import Utilities */
import { renderComponent } from "../../../test/setupTests";

/** -------------------------------------------------------- */

describe("Legal", () => {
   it("Renders ok", () => {
      /** Component context mounting */
      const component = renderComponent(<Legal />);

      /** Main assertion */
      expect(component).not.toBe(null);
   });

   it("Renders main component wrapper", () => {
      /** Component context mounting */
      const { getByTestId } = renderComponent(<Legal />);

      /** Main assertion */
      const component = getByTestId("legal-page");
      expect(component).toBeInstanceOf(HTMLElement);
   });

   it("Matches snapshot", () => {
      /** Component context mounting */
      const { getByTestId } = renderComponent(<Legal />);

      /** Main assertion */
      const component = getByTestId("legal-page");
      expect(component).toMatchSnapshot();
   });
});
