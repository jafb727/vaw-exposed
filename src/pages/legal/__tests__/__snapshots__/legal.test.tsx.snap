// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Legal Matches snapshot 1`] = `
<div
  data-testid="legal-page"
>
  Legal
</div>
`;
